lib/mayaUsd/ufe/UsdContextOps.cpp

```diff
@@ -1249,10 +1249,22 @@ Ufe::UndoableCommand::Ptr UsdContextOps::doOpCmd(const ItemPath& itemPath)
         return std::make_shared<UnbindMaterialUndoableCommand>(fItem->prim());
 #if UFE_PREVIEW_VERSION_NUM >= 4010
     } else if (itemPath.size() == 3u && itemPath[0] == kAssignNewMaterialItem) {
+        Ufe::SceneItemList sceneItems;
+        // As per UX' wishes, we add the item that was right-clicked,
+        // regardless of its selection state.
         if (fItem) {
-            return std::make_shared<InsertChildAndSelectCommand>(
-                UsdUndoAssignNewMaterialCommand::create(fItem, itemPath[2]));
+            sceneItems.push_back(fItem);
         }
+        const Ufe::GlobalSelection::Ptr selection = Ufe::GlobalSelection::get();
+        for (const auto& item : *selection) {
+            const UsdSceneItem::Ptr usdItem = std::dynamic_pointer_cast<UsdSceneItem>(item);
+            if (!usdItem || usdItem == fItem) {
+                continue;
+            }
+            sceneItems.push_back(usdItem);
+        }
```

Comment:
This could be simplified by just using a Ufe Selection (which internally uses a SceneItemList).
```suggestion
        auto sceneItems = Ufe::GlobalSelection::get();
        sceneItems.append(fItem);
```

The Ufe Selection append will only add the item if its not already there. If you really need the item at the front then you can use insert instead. Then just pass this selection to the UsdUndoAssignNewMaterialCommand.

Reply:
No you cannot get the SceneItemList out of it. There are iterators for looping thru. But I would use Ufe::Selection not Ufe::GlobalSelection.

Reply:
Were you by chance getting the Ufe Selection into a reference pointer and them modifying it? You would just need to make sure to get a copy of it (using the copy constructor).